libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxVoiceInstructionsPlayer.kt

```diff
@@ -0,0 +1,102 @@
+package com.mapbox.navigation.ui.voice.api
+
+import android.content.Context
+import androidx.annotation.UiThread
+import com.mapbox.navigation.ui.base.api.voice.VoiceInstructionsPlayer
+import com.mapbox.navigation.ui.base.api.voice.VoiceInstructionsPlayerCallback
+import com.mapbox.navigation.ui.base.model.voice.Announcement
+import com.mapbox.navigation.ui.base.model.voice.SpeechState
+import java.util.Queue
+import java.util.concurrent.ConcurrentLinkedQueue
+
+/**
+ * Hybrid implementation of [VoiceInstructionsPlayer] combining [VoiceInstructionsTextPlayer] and
+ * [VoiceInstructionsFilePlayer] speech players.
+ * @property context Context
+ * @property accessToken String
+ * @property language [Locale] language (ISO 639)
+ */
+@UiThread
+class MapboxVoiceInstructionsPlayer(
+    private val context: Context,
+    private val accessToken: String,
+    private val language: String
+) : VoiceInstructionsPlayer {
+
+    private val queue: Queue<SpeechState.ReadyToPlay> = ConcurrentLinkedQueue()
+    // for testing
+    internal var filePlayer: VoiceInstructionsFilePlayer =
+        VoiceInstructionsFilePlayer(context, accessToken, language)
+    // for testing
+    internal var textPlayer: VoiceInstructionsTextPlayer =
+        VoiceInstructionsTextPlayer(context, language)
+    private val clientCallbacks: Queue<VoiceInstructionsPlayerCallback> = ConcurrentLinkedQueue()
+    private var localCallback: VoiceInstructionsPlayerCallback =
+        object : VoiceInstructionsPlayerCallback {
+            override fun onDone(state: SpeechState.DonePlaying) {
+                val announcement = queue.poll()
+                val clientCallback = clientCallbacks.poll()
```

Comment:
Wondering if it wouldn't be easier to have a data object that contains both `announcement` and `clientCallback` and maintain only one queue.